libs/server/Objects/List/ListObjectImpl.cs

```diff
@@ -418,5 +419,211 @@ private void ListSet(ref ObjectInput input, ref SpanByteAndMemory output)
                 output.Length = (int)(output_currptr - output_startptr);
             }
         }
+
+        private void ListPosition(ref ObjectInput input, ref SpanByteAndMemory output)
+        {
+            var element = input.parseState.GetArgSliceByRef(input.parseStateStartIdx).ReadOnlySpan;
+            input.parseStateStartIdx++;
+
+            var isMemory = false;
+            MemoryHandle ptrHandle = default;
+            var output_startptr = output.SpanByte.ToPointer();
+            var output_currptr = output_startptr;
+            var output_end = output_currptr + output.Length;
+            var count = 0;
+            ObjectOutputHeader outputHeader = default;
+
+            try
+            {
+                if (!ReadListPositionInput(ref input, out var rank, out count, out var maxlen, out var error))
+                {
+                    while (!RespWriteUtils.WriteError(error, ref output_currptr, output_end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref output_startptr, ref ptrHandle, ref output_currptr, ref output_end);
+                    return;
+                }
+
+                if (count < 0)
+                {
+                    while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER, ref output_currptr, output_end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref output_startptr, ref ptrHandle, ref output_currptr, ref output_end);
+                    return;
+                }
+
+                if (maxlen < 0)
+                {
+                    while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER, ref output_currptr, output_end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref output_startptr, ref ptrHandle, ref output_currptr, ref output_end);
+                    return;
+                }
+
+                if (rank == 0)
+                {
+                    while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER, ref output_currptr, output_end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref output_startptr, ref ptrHandle, ref output_currptr, ref output_end);
+                    return;
+                }
+
+                count = count == 0 ? list.Count : count;
+                var totalArrayHeaderLen = 0;
+                var lastFoundItemIndex = -1;
+
+                if (count != 1)
```

Comment:
Looks like there should be a distinction between COUNT argument being present or not. As long as the argument is provided, the return should be an array. Currently, COUNT being passed in as 1 and COUNT not passed in are treated the same, but the return type should be an array when COUNT is passed in even with 1. Same with null reply.

Reply:
I missed that part in my testing. Fixed in now